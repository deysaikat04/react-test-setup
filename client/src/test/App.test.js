import React from "react";
import {Provider} from 'react-redux';
import { mount, shallow } from "enzyme";
import { render, screen } from "@testing-library/react";
import App from "../App";
import renderer from "react-test-renderer";
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk';

const mockStore = configureMockStore([thunk]);

const testStoreVal = {
    id:"",
    ipName: "",
    ipId: "",
    ipClass: "",
    ipSize: "",
    ipType: "",
    ipPlaceholder:"",
    ipLabel:"",
    error: ""
};

const storeVal = {
    attributes: testStoreVal
};

const mockStoreVal = mockStore(storeVal);

const rendererCreateComp = () => renderer.create(<Provider store={mockStoreVal}><App/></Provider>);

describe("App", () => {
  it("renders", () => {
    const component = rendererCreateComp();
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});


