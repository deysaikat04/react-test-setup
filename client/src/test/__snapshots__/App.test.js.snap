// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<div
  className="container"
>
  <div
    className="row text-center mt-5"
  >
    <div
      className="col-6"
    >
      <h4>
        Form
      </h4>
      <p>
        Fill the form and submit to generate dynamic Input Field
      </p>
      <form
        onSubmit={[Function]}
      >
        <div
          className="row g-3 align-items-center mb-3"
        >
          <div
            className="col-3"
          >
            <label
              className="col-form-label"
              htmlFor="ipLabel"
            >
              Label
            </label>
          </div>
          <div
            className="col-6"
          >
            <input
              aria-describedby="ipLabel"
              className="form-control"
              id="ipLabel"
              name="ipLabel"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="row g-3 align-items-center mb-3"
        >
          <div
            className="col-3"
          >
            <label
              className="col-form-label"
              htmlFor="ipName"
            >
              Name
            </label>
          </div>
          <div
            className="col-6"
          >
            <input
              aria-describedby="ipName"
              className="form-control"
              id="ipName"
              name="ipName"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="row g-3 align-items-center mb-3"
        >
          <div
            className="col-3"
          >
            <label
              className="col-form-label"
              htmlFor="ipType"
            >
              Type
            </label>
          </div>
          <div
            className="col-6"
          >
            <input
              aria-describedby="ipType"
              className="form-control"
              id="ipType"
              name="ipType"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="row g-3 align-items-center mb-3"
        >
          <div
            className="col-3"
          >
            <label
              className="col-form-label"
              htmlFor="ipClass"
            >
              Class
            </label>
          </div>
          <div
            className="col-6"
          >
            <input
              aria-describedby="ipClass"
              className="form-control"
              id="ipClass"
              name="ipClass"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="row g-3 align-items-center mb-3"
        >
          <div
            className="col-3"
          >
            <label
              className="col-form-label"
              htmlFor="ipId"
            >
              ID
            </label>
          </div>
          <div
            className="col-6"
          >
            <input
              aria-describedby="ipId"
              className="form-control"
              id="ipId"
              name="ipId"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="row g-3 align-items-center mb-3"
        >
          <div
            className="col-3"
          >
            <label
              className="col-form-label"
              htmlFor="ipSize"
            >
              Size
            </label>
          </div>
          <div
            className="col-6"
          >
            <input
              aria-describedby="ip-size"
              className="form-control"
              id="ipSize"
              name="ipSize"
              onChange={[Function]}
              required={true}
              type="number"
              value={0}
            />
          </div>
        </div>
        <div
          className="row g-3 align-items-center mb-3"
        >
          <div
            className="col-3"
          >
            <label
              className="col-form-label"
              htmlFor="ipSize"
            >
              Placeholder
            </label>
          </div>
          <div
            className="col-6"
          >
            <input
              aria-describedby="input-placeholder"
              className="form-control"
              id="ipPlaceholder"
              name="ipPlaceholder"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          className="btn btn-dark mt-5"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
    <div
      className="col-6"
    >
      <h4>
        Dynamic Input Field
      </h4>
      <p>
        Based on the form values, the 
        <code>
          input
        </code>
         field will appear below.
      </p>
      
    </div>
  </div>
</div>
`;
